dashboard/app/mailers/pd/workshop_mailer.rb

```diff
@@ -30,12 +30,36 @@ class Pd::WorkshopMailer < ActionMailer::Base
 
   after_action :save_timestamp
 
+  def teacher_survey_reminder(enrollment)
+    @enrollment = enrollment
+    @workshop = enrollment.workshop
```

Comment:
I see that you're following a pattern in this controller, but I believe `workshop` can/should be local, not an instance variable.
```rb
    workshop = enrollment.workshop
```
Note `@enrollment` does need to be an instance variable because it gets used in [`save_timestamp`](https://github.com/code-dot-org/code-dot-org/blob/3b6695948a218a8d933a8d6a0b63b801bc5585c6/dashboard/app/mailers/pd/workshop_mailer.rb#L220) and [`generate_csf_certificate`](https://github.com/code-dot-org/code-dot-org/blob/3b6695948a218a8d933a8d6a0b63b801bc5585c6/dashboard/app/mailers/pd/workshop_mailer.rb#L227) below.

Reply:
Just kidding, I didn't realize `ActionMailer::Base` was sharing its instance variables with the haml views for mail.

Reply:
That's correct. Btw, I realize that I added these 2 functions at the top of the file, which seems odd with the current flow in the file. Will move them down in the next commit.